engine/base_client/search.py

```diff
@@ -47,11 +47,22 @@ def _search_one(cls, query: Query, top: Optional[int] = None):
         end = time.perf_counter()
 
         precision = 1.0
+        recall = 1.0
         if query.expected_result:
             ids = set(x[0] for x in search_res)
-            precision = len(ids.intersection(query.expected_result[:top])) / top
 
-        return precision, end - start
+            # precision equals True Positives / (True Positives + False Positives)
+            # recall equals True Positives / (True Positives + False Negatives)
+            # See https://en.wikipedia.org/wiki/Precision_and_recall
+            true_positives = len(ids.intersection(query.expected_result[:top]))
+            precision = (
+                true_positives / top
+            )  # 1 means that the results consist of expected elements.
+            recall = true_positives / len(
```

Comment:
The formal for recall is correct but I wonder how useful it is when we are doing a top-k search.

If we specify `top=5` and there are 100 neighbors in `query.expected_result`. The recall will always be at most `0.05` which doesn't seem to be a useful metric?